### PR TITLE
Fix javadoc

### PR DESCRIPTION
**Pull request by pvalsecc**
A few fronts:
* have javadoc warnings treated as errors
* fix the javadoc warnings
* for not going mad, disable the mandatory javadoc on constants
* remove the error message about MANIFEST.MF not found when not in a
  real servlet
* remove the static fields from the input parameters documentation

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/VersionInfo.java [hunk 1]
```diff
@@ -64,7 +66,8 @@ private String getValue(final String name) {
     }
 
     private Attributes getAttributes() {
-        if (this.servletContext == null) {
+        if (this.servletContext == null ||
+                this.servletContext.getClass().getSimpleName().startsWith("Mock")) {
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/VersionInfo.java]: same above

File: core/src/main/java/org/mapfish/print/VersionInfo.java [hunk 2]
```diff
@@ -37,7 +37,9 @@
     @PostConstruct
     public final void init() {
         this.attrs = getAttributes();
-        LOGGER.warn("Starting print version {} ({})", getVersion(), getGitHash());
+        if (this.attrs != null) {
+            LOGGER.warn("Starting print version {} ({})", getVersion(), getGitHash());
+        }
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/VersionInfo.java]: What's the relation with the javadoc?

File: docs/src/main/groovy/GenerateDocs.groovy [hunk 3]
```diff
@@ -266,17 +268,19 @@ class GenerateDocs {
...
+                def annotations = att.getAnnotations().collect { it.toString() }
+                def rec = new Detail([
+                        title      : att.name,
+                        desc       : desc,
+                        required   : required,
+                        annotations: annotations
+                ])
+
+                details << rec
+            }
```
Comment (sbrunner) [inline docs/src/main/groovy/GenerateDocs.groovy]: same again :-)

File: core/src/main/java/org/mapfish/print/VersionInfo.java (hunk 2 above)
Comment (pvalsecc) [reply core/src/main/java/org/mapfish/print/VersionInfo.java]: See point 4 in the commit comment

File: docs/src/main/groovy/GenerateDocs.groovy (hunk 3 above)
Comment (pvalsecc) [reply docs/src/main/groovy/GenerateDocs.groovy]: See point 5

File: core/src/main/java/org/mapfish/print/VersionInfo.java (hunk 1 above)
Comment (pvalsecc) [reply core/src/main/java/org/mapfish/print/VersionInfo.java]: See point 4

Comment (pvalsecc) [reply core/src/main/java/org/mapfish/print/VersionInfo.java]: When generating the javadoc, I was annoyed by a huge stacktrace. I hate stacktraces when I'm generating javadoc. ;-)

Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/VersionInfo.java]: Relay nice, many thanks :-)

**Pull request merged**